src/Symfony/Component/Translation/Tests/Loader/XliffFileLoaderTest.php

```diff
@@ -149,4 +149,21 @@ public function testLoadNotes()
         // message with empty target
         $this->assertEquals(array('notes' => array(array('content' => 'baz'), array('priority' => 2, 'from' => 'bar', 'content' => 'qux'))), $catalogue->getMetadata('key', 'domain1'));
     }
+
+    public function testLoadVersion2()
+    {
+        $loader = new XliffFileLoader();
+        $resource = __DIR__.'/../fixtures/resources-2.0.xlf';
+        $catalogue = $loader->load($resource, 'en', 'domain1');
+
+        $this->assertEquals('en', $catalogue->getLocale());
+        $this->assertEquals(array(new FileResource($resource)), $catalogue->getResources());
+        $this->assertSame(array(), libxml_get_errors());
+
+        $domains = $catalogue->all();
+        $this->assertCount(3, $domains['domain1']);
+
```

Comment:
please add ``$this->assertContainsOnly('string', $catalogue->all('domain1'));`` as in ``testLoad`` to be sure that the Xliff2 implementation does not get broken in the future by returning nodes without casting them (as done recently for Xliff 1)